### DISCUSSION TITLE
Can't view BrainVISA history

### DISCUSSION THREAD

**Discussion by uthoprakarn** [brainvisa/brainvisa.github.io#104] [Q&A]
Hi all,

I have activated the history for one of my databases, using the "activate_history" option when adding the database, but I can't view its content even though the history files exist.

I have tried through the "Data Management" > "Database browser" process, by right-clicking on a file of the database and selecting "Show history": this crashes BrainVISA with the following error:

```
Traceback (most recent call last):
  File "/casa/host/build/python/brainvisa/data/qt4gui/hierarchyBrowser.py", line 487, in menuHistoryEvent
    parent=self)
  File "/casa/host/build/python/brainvisa/data/qt4gui/history.py", line 137, in __init__
    minf = readMinf(bvproc_file)
  File "/casa/host/build/python/soma/minf/api.py", line 301, in readMinf
    exceptions=exceptions))
  File "/casa/host/build/python/soma/minf/api.py", line 258, in iterateMinf
    % (source.name, ))
soma.minf.error.MinfError: Invalid minf file: /network/lustre/iss01/cati/dev/thoprakarn/bv_tests/history_book/2022-04-22/3dd0137e-6500-4f18-1a46-980f65944d68.bvproc
```
Any idea on why this doesn't work? 

And are there other ways to view the history without using the "Data Management" > "Database browser" process?

Thank you for you help!!!
Urielle

**Comment by clarafischer** [comment]
Hello,

just to add that I have the same problem. I thought it was me who had made a false manipulation on my bases but on a recent clean one I have the same thing as Urielle. It makes an error on the minf but it's there and you can read it with a text editor without any problem and what's annoying is that it still makes the whole brainvisa crash with a nice "Fatal Python error: Aborted" at the end.
It worked before but I couldn't tell how long it's been broken, I should have made a ticket earlier.

Thank you !
Clara

**Comment by sapetnioc** [comment]
It has been a long time since I last used history related code. To read bvproc files, you can use `readMinf`:

```
from soma.minf.api import readMinf

minf = readMinf('/somewhere/something.bvproc')
```

For history management, there is an `HistoryBook` class in `history.py` but it has no documentation. I would need a database with history acivated and a few records in it in order to help you on how to use it.

**Comment by uthoprakarn** [comment]
Thanks! Unfortunately, `readMinf`doesn't work on the bvproc files that are stored in my "history_book" folder.
I get the following error:
```
Invalid minf file: /bv_tests/history_book/2022-05-03/59110d38-d2c5-349f-2217-69095e1acab6.bvproc
```

**Comment by sapetnioc** [comment]
Oh. So I probably told you something wrong about using `readMinf()`. I will have to take a closer look to code an to create some bvprocs to test.

**Reply by uthoprakarn** [reply]
Ok, thanks! Let me know if I can help with anything :blush:

**Comment by c-langlet** [comment]
@sapetnioc You are right `readMinf()` is used to read the `.bvproc` file. The error actually comes from the `iterateMinf()` that is used to determine how to open the file : it compares (using gzip library) the 5 first characters against the string `<xml?` (something like that) but it reads  `b<xml?` so its python2/python3 related. I didn't correct it, because there seemed to be more errors under the hood. Additionally I was not sure if the reading of a `bvproc` was supposed to work with `readMinf`but you answered that ;)
